llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -445,17 +446,18 @@ splitModule(Module &M,
   }
 }
 
-static std::string makeResultFileName(Twine Ext, int I) {
+static std::string makeResultFileName(Twine Ext, int I, bool IsEsimd) {
   const StringRef Dir0 = OutputDir.getNumOccurrences() > 0
                              ? OutputDir
                              : sys::path::parent_path(OutputFilename);
   const StringRef Sep = sys::path::get_separator();
   std::string Dir = Dir0.str();
   if (!Dir0.empty() && !Dir0.endswith(Sep))
     Dir += Sep.str();
-  return (Dir + Twine(sys::path::stem(OutputFilename)) + "_" +
-          std::to_string(I) + Ext)
-      .str();
+  std::string RetStr = Dir + sys::path::stem(OutputFilename).str() + "_";
+  if (IsEsimd)
+    RetStr += "esimd_";
+  return RetStr + std::to_string(I) + Ext.str();
```

Comment:
Am I right that this filename change was only introduced to simplify debugging, i.e. there is no functional need for it?

I have a small concern that a number of functions were changed just to have this prefix added and it is kind of a non-generic solution (like what if we decided to add one more level of device code split for some other feature?)

Reply:
Before my change the code was split into modules and saved to files with the name derived from `OutputFilename` plus sequential index, i.e. `/path/to/file_01.bc`, `/path/to/file_02.bc`, etc. 

Now, when we split ESIMD modules we cannot overwrite already processed SYCL files, this is why we need to change the name of the ESIMD files. We could have:
```
`/path/to/file_01.bc` // sycl
`/path/to/file_02.bc` // sycl
`/path/to/file_esimd_01.bc` // esimd
`/path/to/file_esimd_02.bc` // esimd
```
or
```
`/path/to/file_01.bc` // sycl
`/path/to/file_02.bc` // sycl
`/path/to/file_03.bc` // esimd
`/path/to/file_04.bc` // esimd
```
I choose the first scheme, since, as you said, it will be easier to find ESIMD modules, and the names of the files do not affect functionality. But I'm OK to change if you think the second is better. However, I think we need to propagate the `IsEsimd` flag (or other parameter) regardless of the naming scheme.

We can introduce a class that will keep the state. Then all the propagation will go away. But we can do that later in further patches.

Reply:
> I choose the first scheme, since, as you said, it will be easier to find ESIMD modules, and the names of the files do not affect functionality. But I'm OK to change if you think the second is better. 

I don't have a strong opinion here, so, the proposed way looks fine to me. We can hear from @kbobrovs here to have another opinion

> However, I think we need to propagate the IsEsimd flag (or other parameter) regardless of the naming scheme.

Right, but from functional point of view it is only needed in `processOneModule`

Reply:
> We can introduce a class that will keep the state. Then all the propagation will go away. But we can do that later in further patches.

Personally, I'm not a fun of such states, because it makes it harder to understand where particular value is came from and how the call graph looks like

Reply:
Current scheme with `_esimd` suffix seems clearer to me. Users don't care, but devs will appreciate it.